src/main/webapp/dev/js/main/instructorFeedbackEdit.js

```diff
@@ -226,29 +226,26 @@ function checkFeedbackQuestion(form) {
         return false;
     }
 
-    if ($(form).find(`[name=${ParamsNames.FEEDBACK_QUESTION_TYPE}]`).val() === 'RANK_OPTIONS') {
-        if ($(form).find(`[name=${ParamsNames.FEEDBACK_QUESTION_RANKMINOPTIONSCHECKBOX}]:checked`).val() === 'on'
-                && $(form).find(`[name=${ParamsNames.FEEDBACK_QUESTION_RANKMINOPTIONSTOBERANKED}]`).val() === '') {
+    if ($(form).find(`[name=${ParamsNames.FEEDBACK_QUESTION_TYPE}]`).val() === 'RANK_OPTIONS'
```

Comment:
The condition is too complex?

Reply:
If not it'll be better to split it? But the code will look duplicated as you commented previously

Reply:
If you extract out a function and call it with the necessary parameters would that work?

Reply:
👍. How do we decide when extracting out a function is appropriate or not? Sometimes adding a function to de-duplicate 1 logic might be overdoing. Just want to check this as a general rule of thumb to take note.

Reply:
The problem with duplicate logic is that one small error and it will be hard for the reader to realize. So I don't think it is ever overkill to extract a function to de-duplicate logic (except perhaps in tests).

So it seems that as long as the objective is to make the code as readable as possible, we don't have to care whether it is extracted into a function or not.

Is there a specific case that you were thinking why it might be overkill?